### DISCUSSION TITLE
[LiveComponent] Emitting Event to specific instance

### DISCUSSION THREAD

**Discussion by tito10047** [symfony/ux#2916] [Questions & Answers]
Hi, 
I have 10 instances of same component in page with live prop $id. I want emit event to only one instacne of this component with specific $id.

Is something like this possible?

**Comment by smnandre** [comment]
It is not possible no, but they maybe can "do" something if they match the id of the event ?

**Comment by tito10047** [comment]
Yes, bud this is trigered reload all 10 instances and i want avoid it because render this component is heavy. Maybe not heavy, but now I have 60 instances on page, Ind i need reload only one.

**Comment by smnandre** [comment]
As Event are really just "strings", did you try to emit something like "foo_42" where 42 is the id in question, and listen to this event in your component 42 ?

**Comment by tito10047** [comment]
Im still not understand what you mean..
I can listen to specific event "foo" like:
```
#[LiveListener('foo')]
public function foo(#[LiveArg] int $id){}
```
I can't add id to listener name  like ```#[LiveListener('foo_42')]```. And if all instacnes of compoent have same listener name, then all will be listened for event, that means for all instances of component will be called http request individualy

**Comment by smnandre** [comment]
So there is nothing like that in LiveComponent sorry. The compote works around events, delegation or "in-scope" actions.

**Comment by smnandre** [comment]
Depending on "who" emit the initial event you're talking about... 

... you could listen on front-end with a custom stimulus controller on your component instances, where you would check the "id" and then dispatch a "emitSelf" event.

Is this the same scenario that the other issue ?

**Comment by tito10047** [comment]
Yesm this custom stimulus controller is  good enough for me. thanks
